src/txdb.cpp

```diff
@@ -567,10 +567,13 @@ bool CBlockTreeDB::LoadBlockIndexGuts(
 
                 // ZIP 221 consistency checks
                 if (chainParams.GetConsensus().NetworkUpgradeActive(pindexNew->nHeight, Consensus::UPGRADE_HEARTWOOD)) {
-                    if (pindexNew->hashLightClientRoot != pindexNew->hashChainHistoryRoot) {
+                    // Don't enforce consistency check for headers that aren't necessarily
+                    // Heartwood-aware (such as headers we receive from non-upgraded peers
+                    // immediately after Heartwood activation).
+                    if (pindexNew->IsValid(BLOCK_VALID_CONSENSUS) && pindexNew->hashLightClientRoot != pindexNew->hashChainHistoryRoot) {
```

Comment:
After pairing with @lindanlee on reviewing this, we were concerned about this check possibly not being enforced for blocks that are invalid for some _other_ reason than what happened on testnet yesterday specifically. If it's possible for blocks to be marked invalid temporarily and then become valid, that might be used to circumvent the check. We didn't have time to dig into the surrounding code and find out if there's really an issue so I'm leaving this note here to remind me when I/we complete the review.